doc/howto/autoinstall-validation.rst

```diff
@@ -0,0 +1,249 @@
+
+Common Mistake #3
+^^^^^^^^^^^^^^^^^
+
+Another possible mistake is to think that the autoinstall config on the installation media is a cloud-config datasource (it is not):
+
+.. tabs::
+
+   .. tab:: Validation output
+
+      When providing the autoinstall configuration using the top-level ``autoinstall`` keyword format, the installer will verify there are no other top-level keys:
+
+      .. code:: none
+
+         $ ./scripts/validate-autoinstall-user-data.py --no-expect-cloudconfig <path-to-config>
+         error: subiquity/load_autoinstall_config/read_config: autoinstall.yaml is not a valid cloud config datasource.
+         No other keys may be present alongside 'autoinstall' at the top level.
+         Malformed autoinstall in 'top-level keys' section
+         Failure: The provided autoinstall config did not validate succesfully
+
+   .. tab:: Faulty config
+
+      The following config contains cloud-config directives when it is not expected to contain any:
+
+      .. code:: none
+
+         #cloud-config
+
+         # some cloud-config diretives
+
+         autoinstall:
+            # autoinstall directives
+
+
+
+Debugging errors
+^^^^^^^^^^^^^^^^
+
+By default, the validation script has low verbosity output:
+
+.. code:: none
+
+   Malformed autoinstall in 'version or interactive-sections' section
+   Failure: The provided autoinstall config did not validate succesfully
+
+However, you can increase the output level by successively passing the ``-v`` flag. At maximum verbosity, the validation script will report errors the same way they are reported at runtime.  This is great for inspecting issues in cases where the short error message isn't yet specific enough to be useful and can be used to inspect specific JSON schema validation errors.
+
+
+.. code:: none
+
+   $ ./scripts/validate-autoinstall-user-data.py autoinstall.yaml  -vvv
+   start: subiquity/load_autoinstall_config:
+   start: subiquity/load_autoinstall_config/read_config:
+   finish: subiquity/load_autoinstall_config/read_config: SUCCESS:
+   start: subiquity/Reporting/load_autoinstall_data:
+   finish: subiquity/Reporting/load_autoinstall_data: SUCCESS:
+   start: subiquity/Error/load_autoinstall_data:
+   finish: subiquity/Error/load_autoinstall_data: SUCCESS:
+   start: subiquity/core_validation:
+   finish: subiquity/core_validation: FAIL: Malformed autoinstall in 'version or interactive-sections' section
+   finish: subiquity/load_autoinstall_config: FAIL: Malformed autoinstall in 'version or interactive-sections' section
+   Malformed autoinstall in 'version or interactive-sections' section
+   Traceback (most recent call last):
+     File "/home/cpete/software/subiquity/scripts/../subiquity/server/server.py", line 654, in validate_autoinstall
```

Comment:
I'd typically drop the "/home/cpete/software/subiquity" prefix from the tracebacks but it's up to you

Reply:
Good catch, I hadn't thought about that. I replaced the real prefix with ellipses. Does that seem better?